kube-controllers/pkg/controllers/node/ipam.go

```diff
@@ -568,16 +574,26 @@ func (c *ipamController) checkEmptyBlocks() error {
 			continue
 		}
 
+		// During a Flannel migration, we can only migrate blocks affined to nodes that have already undergone the migration
+		migrating, err := c.nodeIsBeingMigrated(node)
+		if err != nil {
+			logc.Warn("Couldn't find node affined to empty block, skipping affinity release")
+			continue
+		}
+		if migrating {
+			logc.Info("Node affined to block is currently undergoing a migration from Flannel, skipping affinity release")
```

Comment:
I believe you're missing a `continue` in this block

Reply:
thanks.

Reply:
I haven't tested the migration again.
I can probably test it tomorrow however.
Do you have tests for the migration yourself?